libs/sdk-ui-charts/src/highcharts/chartTypes/_util/common.ts

```diff
@@ -166,8 +166,26 @@ export const unwrap = (wrappedObject: unknown): any => {
     return wrappedObject[Object.keys(wrappedObject)[0]];
 };
 
-export function percentFormatter(value: number): string {
-    return isNil(value) ? "" : `${parseFloat(value.toFixed(2))}%`;
+const getNumberWithGivenDecimals = (value: number, decimalNumbers: number) =>
+    decimalNumbers === 0 ? `${Math.round(value)}%` : `${parseFloat(value.toFixed(decimalNumbers))}%`;
+
+export function percentFormatter(value: number, format?: string): string {
+    let numberOfDecimals = 2;
+
+    if (format && !isNil(value)) {
+        const splittedFormat = format.split(".");
+        const isPercent = format.trim().slice(-1) === "%";
+
+        if (isPercent) {
+            numberOfDecimals = 0;
+
+            if (splittedFormat.length > 1) {
+                Array.from(splittedFormat[1]).forEach((letter) => letter === "0" && numberOfDecimals++);
```

Comment:
Why do you need `Array.from` here? What is it supposed to achieve? 

Again, opinionated suggestion: rather than `forEach` modifying variable as side-effectI would rather use `reduce`: 

`return splittedFormat[1].reduce((decimalsNumber, letter) => letter === "0" ? decimalsNumber + 1 : decimalsNumber, 0)`

Possibly the method would be more readable if you would keep indentation low and use immediate return when you know the result and do not encapsulate `if` branches into each other. The function is easier to reason about and to write tests to as you see the possibilities right away.

Again, that this as an opinion and suggestion. You don't need to change it as it is not wrong.